docker/development/Dockerfile

```diff
@@ -0,0 +1,65 @@
+FROM node:alpine
+
+LABEL mantainer "Diego Dorgam <diego.dorgam@rocket.chat>"
+
+ENV HUBOT_LANG='en'                                                  \
+    HUBOT_CORPUS='training_data/corpus.yml'                                        \
+    HUBOT_ADAPTER=rocketchat                                         \
+    HUBOT_OWNER=RocketChat                                           \
+    HUBOT_NAME=HubotNatural                                          \
+    HUBOT_DESCRIPTION="Processamento de linguagem natural com hubot" \
+    HUBOT_LOG_LEVEL=debug                                            \
+    ROCKETCHAT_URL=http://rocketchat:3000                            \
+    ROCKETCHAT_ROOM=GENERAL                                          \
+    ROCKETCHAT_USER=chatbot                                          \
+    ROCKETCHAT_PASSWORD=@12345@                                      \
+    ROCKETCHAT_AUTH=password                                         \
+    RESPOND_TO_DM=true                                               \
+    RESPOND_TO_LIVECHAT=true                                         \
+    RESPOND_TO_EDITED=true                                           \
+    LIVECHAT_DEPARTMENT_ID=null                                      \
+    LISTEN_ON_ALL_PUBLIC=true
+
+RUN apk --update add --no-cache git make gcc g++ python python3 && \
+    addgroup -S hubotnat && adduser -S -g hubotnat hubotnat
+
+USER node
+
+RUN mkdir /home/node/.npm-global && \
+    chown -R node:node /home/node/.npm-global
+
+ENV PATH=/home/node/.npm-global/bin:$PATH \
+    NPM_CONFIG_PREFIX=/home/node/.npm-global
+
+RUN npm install -g yo generator-hubot
+
+WORKDIR /home/hubotnat/bot
+
+USER root
+
+ADD ./docker/development/bot_config.py /home/hubotnat/bot_config.py
+
+RUN mkdir -p /home/hubotnat/.config/configstore                             && \
+    echo "optOut: true" > /home/hubotnat/.config/configstore/insight-yo.yml && \
+    chown -R hubotnat:hubotnat /home/hubotnat
+
+USER hubotnat
+
+RUN yo hubot --adapter ${HUBOT_ADAPTER}         \
+             --owner ${HUBOT_OWNER}             \
+             --name ${HUBOT_NAME}               \
+             --description ${HUBOT_DESCRIPTION} \
+             --defaults --no-insight         && \
+    rm /home/hubotnat/bot/external-scripts.json \
+    /home/hubotnat/bot/scripts/example.coffee   \
+    /home/hubotnat/bot/hubot-scripts.json
+
+COPY ["package.json", "/home/hubotnat/bot/"]
+
+ADD scripts/ /home/hubotnat/bot/scripts/
+
+RUN pip3 install requests --user
+
+ADD training_data/ /home/hubotnat/bot/training_data
+
+ENTRYPOINT python3 /home/hubotnat/bot_config.py && /home/hubotnat/bot/bin/hubot -a rocketchat
```

Comment:
The docker image must be always as slim as possible
Installing python3 for running the bot config is not a good pratice for 2 reasons, we might not need to run the bot_config.py script in every case, and we wouldn't need python3 inside the image for anything else.
We could execute the python3 script in a different container, just for a bot configs setup. Maybe even more specialized in setup functions.
What you think about it?

Reply:
That's true, I agree that installing python only to use on script execution is not a good idea.
But I don't know if it is worth to maintain another service only to run the script that configures bot.

The solution I can think is to remove the script execution from Hubot-Natural Dockerfile, place the script on repository and indicate its purpose on documentation. That way, the user runs it on his own machine, if they want to configure automatically the bot. 
That approach would simplify too my contribution, once would not be necessary to have neither "development.yml" and "production.yml" files, norr the documentation about the use of its files. So I could go back to the scheme using only "docker-compose.yml".
I only would mantain the 'docker' directory, and the docker files(Dockerfile, docker-compose.yml) organized inside of it.
That would be good?
Sorry I didn't answer before.

Reply:
Pull Request updated according to previous comment. The organization is better, and the documentation is more simple now.